### ISSUE TITLE
Allow for visitor ordering or multi-pass configurations

### ISSUE THREAD

**Issue by riskygit** [parcel-bundler/lightningcss#459]
It seems all the issues I've encountered with lightningcss involve the inability to visit a token more than once or to control visitor ordering. If you set a no-op FunctionExit visitor, a function will be visited after its arguments, but then it won't get minified because it has already been visited. Perhaps it should be possible to set a FunctionExit visitor that does nothing but imply a function should be evaluated after its arguments without overriding the function's behavior... or this could be the default (wherein functions are always evaluated after arguments). Or a visitor to a token could be configured to occur before other tokens.

**Comment by devongovett** [comment]
Could you give an example of what you're trying to accomplish?

**Comment by riskygit** [comment]
Yes, functions like `color-mix` that can be evaluated at compile-time don't wait for evaluation of inner functions. If you have a custom function, `cvar`, defined via a `Function` visitor and it returns a `CssColor` token, the following example _will not get reduced_:
```
.test {
  color: color-mix(in srgb, cvar(--blue) 50%, red 40%);
}
```
As far as I'm aware, there is no way to have lightningcss evaluate `cvar` before `color-mix` without overriding the implementation of `color-mix` (which would involve defining a `color-mix` `FunctionExit` visitor.

**Comment by devongovett** [comment]
`color-mix` in particular is problematic because it is never stored. It is converted to a `CssColor` [during parsing](https://github.com/parcel-bundler/lightningcss/blob/78ab48f5a0cadcea48c6c1fb270b28df3631eb7c/src/values/color.rs#L2812) when all of the parameters are known, and left unchanged when it is unable to be fully parsed. This means that custom visitors will never see it when the values are all static, and it will never be re-processed when non-static values (e.g. variables) are replaced at build time. This was implemented before we had custom visitors so it was fine to do at that point, but we'd need to rework it a bit to support this.

**Comment by riskygit** [comment]
Anything would be an improvement over my current solution of doing a regex replacement on the whole CSS doc before passing it to `transform`. Unfortunately, even if I populate a map of data first (like https://github.com/parcel-bundler/lightningcss/blob/master/node/test/visitor.test.mjs#L241 or https://github.com/parcel-bundler/lightningcss/blob/master/node/test/visitor.test.mjs#L294), the tokens are useless outside lightningcss visitors. It's all delicate parsing and regex replacements.
